src/vs/vscode.d.ts

```diff
@@ -1812,6 +1812,56 @@ declare module 'vscode' {
 	 */
 	export type ProviderResult<T> = T | undefined | null | Thenable<T | undefined | null>;
 
+	/**
+	 * Scope of a code action.
+	 *
+	 * Specifies the type of a code action. A scope is a hierarchical list of identifiers separated by `.`, e.g. `"refactoring.extract.function"`.
+	 */
+	export class CodeActionScope {
```

Comment:
Maybe `CodeActionKind` is the better name because this is about the 'character' of a code action and because we have been using ...Kind already for similar purposes...